clients/drcachesim/common/options.cpp

```diff
@@ -42,6 +42,20 @@ droption_t<bool> op_offline
  "If this option is enabled, trace data is instead written to files in -outdir "
  "for later offline analysis.  No simulator is executed.");
 
+droption_t<unsigned int> op_num_threads
+(DROPTION_SCOPE_CLIENT, "num_threads", 0, "Number of sideline threads for writing traces",
+ "For the offline analysis mode (when -offline is requested), specifies the number "
+ "of sideline threads to be used to write trace files out.  "
+ "Only 0 and 1 is currently supported.  "
+ "0 means synchronized write without any sideline threads.  ");
```

Comment:
IMHO the pool should be on by default, for code simplicity.

I'm not sure about only supporting 1 thread: we need to know that the design here supports many threads, and seeing the stop-profiling scheme below I'm not sure that it does.  Adding >1 thread seems like just a few lines of code: array of events, for loop when touching those events.  It seems better to include multi-thread support now, to ensure the design works (again see below where I'm not sure it does).